subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -743,6 +744,14 @@ static int scan_delegator_add_src(struct bt_conn *conn,
 			     subgroup->metadata_len);
 	}
 
+	if (scan_delegator_cp_cbs != NULL && scan_delegator_cp_cbs->add_source != NULL) {
+		err = scan_delegator_cp_cbs->add_source(conn, state);
+		if (err != BT_ATT_ERR_SUCCESS) {
+			LOG_DBG("add_source callback rejected: 0x%02x", err);
+			return BT_GATT_ERR(err);
```

Comment:
If the application returns e.g. `-1` (for some reason), this will end up being `1` which will return a success to the remote device. 

We could consider doing
```suggestion
			return err > 0 ? BT_GATT_ERR(err) : err;
```

To convert it to an negative value if not already, as a precaution. What do you think?